packages/gatsby-plugin-offline/src/gatsby-browser.js

```diff
@@ -1 +1,40 @@
 exports.registerServiceWorker = () => true
+
+let swNotInstalled = true
+const pathnameResources = []
+
+exports.onPrefetchPathname = ({ pathname, getResourcesForPathname }) => {
```

Comment:
This could use a comment to explain that here we record resources that are prefetched before the sw has been installed and can cache things directly.